packages/xod-project/src/patch.js

```diff
@@ -1391,43 +1391,42 @@ export const checkSpecializationMatchesAbstraction = def(
       );
     }
 
-    const [abstractPatchPins, checkedPatchPins] = R.map(
-      R.compose(
-        R.map(R.sortBy(Pin.getPinOrder)),
-        R.groupBy(Pin.getPinDirection),
-        listPins
-      )
+    const [
+      abstractPatchPinsNumberByDirection,
+      checkedPatchPinsNumberByDirection,
+    ] = R.map(
+      R.compose(R.map(R.length), R.groupBy(Pin.getPinDirection), listPins)
     )([abstractPatch, specializationPatch]);
 
     // check that patch has a proper number of inputs and outputs
     if (
-      abstractPatchPins[CONST.PIN_DIRECTION.INPUT].length !==
-      checkedPatchPins[CONST.PIN_DIRECTION.INPUT].length
+      abstractPatchPinsNumberByDirection[CONST.PIN_DIRECTION.INPUT] !==
```

Comment:
PinsNumber is incorrect. NumberOfPins or PinCount (note, pin is singular in this case)